subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/ContinuousBuildFileWatchingIntegrationTest.groovy

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.watch
+
+import org.gradle.integtests.fixtures.AbstractContinuousIntegrationTest
+import org.gradle.integtests.fixtures.FileSystemWatchingFixture
+import org.gradle.integtests.fixtures.executer.GradleContextualExecuter
+import org.gradle.internal.service.scopes.VirtualFileSystemServices
+
+import static org.junit.Assume.assumeFalse
+
+class ContinuousBuildFileWatchingIntegrationTest extends AbstractContinuousIntegrationTest implements FileSystemWatchingFixture {
+
+    def setup() {
+        assumeFalse("No shared state without a daemon", GradleContextualExecuter.noDaemon)
+        assumeFalse("The test manually enables file system watching", GradleContextualExecuter.watchFs)
+
+        executer.requireIsolatedDaemons()
+        executer.beforeExecute {
+            withWatchFs()
+            withVerboseVfsLog()
```

Comment:
NB: this will conflict with my recent changes in #14366.

Reply:
There is no merge conflict, but this code won't work now that the other PR is merged.

Reply:
I adjusted the code to work with current `master`.